cpp/src/cider/include/cider/processor/BatchProcessorContext.h

```diff
@@ -52,9 +59,19 @@ class BatchProcessorContext {
     return buildTableSupplier_;
   }
 
+  void setCrossJoinColumnBatchSupplier(
+      const CrossJoinColumnBatchSupplier& columnBatchSupplier) {
+    columnBatchSupplier_ = columnBatchSupplier;
+  }
+
+  const CrossJoinColumnBatchSupplier& getCrossJoinColumnBatchSupplier() const {
+    return columnBatchSupplier_;
+  }
+
  private:
   std::shared_ptr<CiderAllocator> allocator_;
   HashBuildTableSupplier buildTableSupplier_;
+  CrossJoinColumnBatchSupplier columnBatchSupplier_;
```

Comment:
maybe rename to 'crossBuildTableSupplier' and rename 'buildTableSupplier' to 'hashBuildTableSupplier' sounds better?